Sources/Workspace/Workspace.swift

```diff
@@ -373,20 +385,29 @@ public class Workspace {
         // a valid manifest with name same as the package we are trying to edit.
         if fileSystem.exists(destination) {
             // Get tools version and try to load the manifest.
-            let toolsVersion = try toolsVersionLoader.load(
-                at: destination, fileSystem: fileSystem)
+            guard let toolsVersion = diagnostics.wrap({
+                try toolsVersionLoader.load(at: destination, fileSystem: fileSystem)
+            }) else {
+                return
+            }
 
-            let manifest = try manifestLoader.load(
-                package: destination,
-                baseURL: dependency.repository.url,
-                manifestVersion: toolsVersion.manifestVersion)
+            guard let manifest = diagnostics.wrap({
+                try manifestLoader.load(
+                    package: destination,
+                    baseURL: dependency.repository.url,
+                    manifestVersion: toolsVersion.manifestVersion)
+            }) else {
+                return
+            }
```

Comment:
We now have a helper method to load the manifest in workspace, we should let it accept a diagnostics manager and use that here instead.